### DISCUSSION TITLE
x-trap.noscroll doesn't work on iPhone

### DISCUSSION THREAD

**Discussion by earthboundkid** [alpinejs/alpine#2368] [7. Bugs (archived)]
I can still scroll the background behind my modal even with x-trap.noscroll. I think this is just a general problem with iOS.

**Comment by danddanddand** [comment]
Alpine should probably use https://github.com/willmcpo/body-scroll-lock

**Reply by danddanddand** [reply]
I made a PR for it to support iOS https://github.com/alpinejs/alpine/pull/2369 by using package mentioned above

**Reply by SimoTod** [reply]
It looks like that library does a few more things (for example, the whole touch events sections). 

From a quick check on the internet, the overflow hidden doesn't work on IOS < 13 but it works on recent versions. I wonder if it would be acceptable to detect IOS < 13 and, in that case, just use the position fixed hack as that library does (https://github.com/willmcpo/body-scroll-lock/blob/master/src/bodyScrollLock.js#L120-L123) and see if anyone raise any bugs. You can use Alpine.setStyle which returns a callback to restore the previous element.

Another observation is that the project has 71 issues. For instance, the last one https://github.com/willmcpo/body-scroll-lock/issues/239 would not occur with the current implementation so someone could flag it as a regression. I'm not sure if we are closing ourselves in a corner if we add a dependency on a project that is not regularly maintained (last commit in august).

**Reply by calebporzio** [reply]
Good insights @SimoTod - I agree
